### DISCUSSION TITLE
On documenting the dbt Cloud API

### DISCUSSION THREAD

**Discussion by matt-winkler** [dbt-labs/devhub-writing#49] [Analytics craft]
### Background

API documentation is important for dbt Cloud users who seek to automate deployment and management of their accounts through custom scripting or tooling such as terraform.

### Current State

The API V2 [docs](https://docs.getdbt.com/dbt-cloud/api-v2) have robust information on jobs management with CRUD operations, job triggering, and job run analysis. There are additional undocumented endpoints available in the API that include capabilities such as creating dbt Cloud Projects, git and database connections, environments etc. 

There is an existing [postman collection](https://dbtlabs.postman.co/workspace/Team-Workspace~520c7ac4-3895-4779-8bc3-9a11b5287c1c/collection/12491709-e1e12626-73c3-4dc6-840f-ed59b78b3836?ctx=documentation) which documents many of the the additional endpoints beyond jobs. Note there is reasonable coverage of endpoints in this collection but not exhaustive. 

### Goal

Create a living (comprehensive and up-to-date) resource for dbt Cloud users that illustrates the API endpoints available and expected input / output of requests. 

### Open Questions

As API endpoints are added and evolve, how will this resource be kept up-to-date?
Is there a case for documenting processes that may require calls to multiple API endpoints in succession?

**Comment by jasnonaz** [comment]
Hi Matt - thanks for putting together this tremendous resource!!

The Postman collection looks great and is definitely something we'd want to get into the hands of the community.

Question - for the associated blog / writeup, what do we think is highest leverage? Could imagine a relatively spartan announcement that this is a resource that we have and invite people to check it out, or we could walk through some of the functionality contained within and how orgs might get value out of it. cc @grantwinship-consulting who is going to be helping out with the writeup.

**Reply by matt-winkler** [reply]
Just a quick callout that we should likely share a static json link to the collection with instructions for users to view themselves in the writeup vs. sharing the workspace which they won't be able to access.

I do think it'd be a good idea to overview a little workflow of say, creating an environment (maybe with some environment variables) in the blog post. It may be useful to put this in blogs.getdbt.com to describe that workflow and involve others from product (@schottj maybe) to add on with any comments on future direction here from an infrastructure management standpoint.

**Reply by schottj** [reply]
@matt-winkler I'd be happy to write up some background on jobs, environment, projects, environment variables, and anything ops related on why these endpoints are useful and give some motivation for how folks are building custom workflows with our APIs.

**Comment by krevitt** [comment]
@jasnonaz adding a rough first draft from Matt here, let's convert this to an issue:
https://docs.google.com/document/d/1QrotsBdQSkw1wJKXhsa-Fvwx1HlRone_TxwaxLBLjFU/edit

**Comment by jasnonaz** [comment]
Looks like we're good to go ahead with this - I'll open up an issue.
